File: 1788-maximize-the-beauty-of-the-garden/1788-maximize-the-beauty-of-the-garden.cpp
class Solution {
public:
    int maximumBeauty(vector<int>& flowers) {
        unordered_map<int, int> m;
        int res = INT_MIN;
        int sum = 0;
        for(int b : flowers) {
            if(m.count(b) > 0) res = max(res, sum - m[b] + 2*b);
            if(b > 0) sum += b;
            if(m.count(b) == 0) m[b] = sum;
        }
        return res;
    }
};

File: 1788-maximize-the-beauty-of-the-garden/1788-maximize-the-beauty-of-the-garden.java
class Solution {
    public int maximumBeauty(int[] flowers) {
        Map<Integer, Integer> m = new HashMap<>();
        int res = Integer.MIN_VALUE;
        int sum = 0;
        for (int b : flowers) {
            if (m.containsKey(b)) {
                res = Math.max(res, sum - m.get(b) + 2 * b);
            }
            if (b > 0) {
                sum += b;
            }
            if (!m.containsKey(b)) {
                m.put(b, sum);
            }
        }
        return res;
    }
}